docs/content/develop/run-tests.md

```diff
@@ -1,53 +1,182 @@
-{{< /hint >}}
+## Before you begin
+
+1. [Generate the modified provider(s)](/magic-modules/docs/getting-started/generate-providers/)
+
+
+1. Set up application default credentials for Terraform
+
+    ```bash
+    gcloud auth application-default login
+    export GOOGLE_USE_DEFAULT_CREDENTIALS=TRUE
+    ```
+
+1. Set the following environment variables:
+
+    ```bash
+    export GOOGLE_USE_DEFAULT_CREDENTIALS=TRUE
+    export GOOGLE_PROJECT=PROJECT_ID
+    export GOOGLE_REGION=us-central1
+    export GOOGLE_ZONE=us-central1-a
+    ```
+
+1. Optional: Some tests may require additional variables to be set, such as:
+
+    ```
+    GOOGLE_ORG
+    GOOGLE_BILLING_ACCOUNT
+    ```
+
+{{< tabs "version" >}}
+
+{{< tab "GA Provider" >}}
+
+## Run automated tests
+
+Run unit tests and linters
+
+```bash
+cd $GOPATH/src/github.com/hashicorp/terraform-provider-google
+make test
+make lint
+```
+
+## Run acceptance tests
+
+Run acceptance tests
+
+```bash
+make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool_basic'
+```
+
+TESTARGS allows you to pass [testing flags](https://pkg.go.dev/cmd/go#hdr-Testing_flags) to `go test`. The most important is `-run`, which allows you to limit the tests that get run. There are 2000+ tests, and running all of them takes over 9 hours and requires a lot of GCP quota.
+
+`-run` is regexp-like, so multiple tests can be run in parallel by specifying a common substring of those tests (for example, `TestAccContainerNodePool` to run all node pool tests).
+
+### Debugging tests
+
+Optional: Save verbose test output to a file for analysis.
+
+```bash
+TF_LOG=TRACE make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+```
+
+{{< /tab >}}
+
+{{< tab "Beta Provider" >}}
+
+## Run automated tests
+
+Run unit tests and linters
+
+```bash
+cd $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+make test
+make lint
+```
+
+## Run acceptance tests
+
+Run acceptance tests
+
+```bash
+make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool_basic'
+```
+
+TESTARGS allows you to pass [testing flags](https://pkg.go.dev/cmd/go#hdr-Testing_flags) to `go test`. The most important is `-run`, which allows you to limit the tests that get run. There are 2000+ tests, and running all of them takes over 9 hours and requires a lot of GCP quota.
+
+`-run` is regexp-like, so multiple tests can be run in parallel by specifying a common substring of those tests (for example, `TestAccContainerNodePool` to run all node pool tests).
+
+### Debugging tests
+
+Optional: Save verbose test output to a file for analysis.
+
+```bash
+TF_LOG=TRACE make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+You can also debug tests with [Delve](https://github.com/go-delve/delve):
```

Comment:
Could this be phrased as an optional step in the tabs above?

Reply:
Done.